csrc/multidevice/communicator.cpp

```diff
@@ -68,7 +71,10 @@ bool parseEnv(
   if (!env) {
     env = std::getenv("WORLD_SIZE");
     if (!env) {
-      return false;
+      env = std::getenv("SLURM_NTASKS");
```

Comment:
Since now it hits the magic number of 3 :), would you mind making a helper to iterate through a list of environmental variables and stop at the first one that succeeds? For example,

```
char* env = tryReadEnv({"OMPI_COMM_WORLD_SIZE", "WORLD_SIZE", "SLURM_NTASKS"});
```

Reply:
makes sense!